src/components/Api.js

```diff
@@ -0,0 +1,118 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    (this._baseUrl = baseUrl), (this._headers = headers);
+  }
+
+  async validateAndParse(res) {
+    if (res.ok) {
+      return await res.json();
+    } else {
+      return Promise.reject();
+    }
+  }
+
+  async getInitialCards() {
+    try {
+      const res = await fetch(`${this._baseUrl}/cards`, {
+        headers: this._headers,
+      });
+      return this.validateAndParse(res);
+    } catch (err) {

```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request and this end is in `index.js`   after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.